File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/campaign.dart
class Campaign {
  final int campaignId;
  final int organisationProfileId;
  final String organisationName;
  final int? categoryId;
  final String? categoryName;
  final String title;
  final String? description;
  final DateTime? startDate;
  final DateTime? endDate;
  final double targetAmount;
  final double currentAmount;
  final String status;
  final DateTime createdAt;
  final int donationCount;

  Campaign({
    required this.campaignId,
    required this.organisationProfileId,
    required this.organisationName,
    this.categoryId,
    this.categoryName,
    required this.title,
    this.description,
    this.startDate,
    this.endDate,
    required this.targetAmount,
    required this.currentAmount,
    required this.status,
    required this.createdAt,
    required this.donationCount,
  });

  factory Campaign.fromJson(Map<String, dynamic> json) => Campaign(
        campaignId: json['campaignId'],
        organisationProfileId: json['organisationProfileId'],
        organisationName: json['organisationName'],
        categoryId: json['categoryId'],
        categoryName: json['categoryName'],
        title: json['title'],
        description: json['description'],
        startDate: json['startDate'] != null ? DateTime.parse(json['startDate']) : null,
        endDate: json['endDate'] != null ? DateTime.parse(json['endDate']) : null,
        targetAmount: (json['targetAmount'] as num).toDouble(),
        currentAmount: (json['currentAmount'] as num).toDouble(),
        status: json['status']?.toString() ?? '',
        createdAt: DateTime.parse(json['createdAt']),
        donationCount: (json['donationCount'] as num?)?.toInt() ?? 0,
      );
}

File: HeartForCharity/HeartForCharity.Services/Database/Campaign.cs
using HeartForCharity.Model.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HeartForCharity.Services.Database
{
    public class Campaign
    {
        [Key]
        public int CampaignId { get; set; }

        [Required]
        [ForeignKey(nameof(OrganisationProfile))]
        public int OrganisationProfileId { get; set; }

        [ForeignKey(nameof(Category))]
        public int? CategoryId { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = null!;

        [MaxLength(4000)]
        public string? Description { get; set; }

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TargetAmount { get; set; }

        [Required]
        public CampaignStatus Status { get; set; } = CampaignStatus.Active;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? DeletedAt { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal CurrentAmount { get; set; } = 0;

        public virtual OrganisationProfile OrganisationProfile { get; set; } = null!;
        public virtual Category? Category { get; set; }
        public virtual ICollection<Donation> Donations { get; set; } = new List<Donation>();
        public virtual ICollection<CampaignMedia> CampaignMedias { get; set; } = new List<CampaignMedia>();
    }
}